tests/baselines/reference/tsbuild/declarationEmit/initial-build/when-declaration-file-is-referenced-through-triple-slash-and-uses-baseUrl.js

```diff
@@ -0,0 +1,290 @@
+interface Array<T> { length: number; [n: number]: T; }
+interface ReadonlyArray<T> {}
+declare const console: { log(msg: any): void; };
+
+//// [/src/solution/src/common/nominal.ts]
+/// <reference path="./types.d.ts" />
+export declare type Nominal<T, Name extends string> = MyNominal<T, Name>;
+
+//// [/src/solution/src/common/tsconfig.json]
+{"extends":"../../tsconfig.base.json","compilerOptions":{"composite":true,"baseUrl":"./"},"include":["./nominal.ts"]}
+
+//// [/src/solution/src/common/types.d.ts]
+declare type MyNominal<T, Name extends string> = T & {
+    specialKey: Name;
+};
+
+//// [/src/solution/src/subProject/index.ts]
+import { Nominal } from '../common/nominal';
+export type MyNominal = Nominal<string, 'MyNominal'>;
+
+//// [/src/solution/src/subProject/tsconfig.json]
+{"extends":"../../tsconfig.base.json","compilerOptions":{"composite":true},"references":[{"path":"../common"}],"include":["./index.ts"]}
+
+//// [/src/solution/src/subProject2/index.ts]
+import { MyNominal } from '../subProject/index';
+const variable = {
+    key: 'value' as MyNominal,
+};
+export function getVar(): keyof typeof variable {
+    return 'key';
+}
+
+//// [/src/solution/src/subProject2/tsconfig.json]
+{"extends":"../../tsconfig.base.json","compilerOptions":{"composite":true},"references":[{"path":"../subProject"}],"include":["./index.ts"]}
+
+//// [/src/solution/src/tsconfig.json]
+{"compilerOptions":{"composite":true},"references":[{"path":"./subProject"},{"path":"./subProject2"}],"include":[]}
+
+//// [/src/solution/tsconfig.base.json]
+{"compilerOptions":{"rootDir":"./","outDir":"lib"}}
+
+//// [/src/solution/tsconfig.json]
+{"compilerOptions":{"composite":true},"references":[{"path":"./src"}],"include":[]}
+
+
+
+Output::
+/lib/tsc --b /src/solution/tsconfig.json --verbose
+[[90m12:00:00 AM[0m] Projects in this build: 
+    * src/solution/src/common/tsconfig.json
+    * src/solution/src/subProject/tsconfig.json
+    * src/solution/src/subProject2/tsconfig.json
+    * src/solution/src/tsconfig.json
+    * src/solution/tsconfig.json
+
+[[90m12:00:00 AM[0m] Project 'src/solution/src/common/tsconfig.json' is out of date because output file 'src/solution/lib/src/common/nominal.js' does not exist
+
+[[90m12:00:00 AM[0m] Building project '/src/solution/src/common/tsconfig.json'...
+
+[[90m12:00:00 AM[0m] Project 'src/solution/src/subProject/tsconfig.json' is out of date because output file 'src/solution/lib/src/subProject/index.js' does not exist
+
+[[90m12:00:00 AM[0m] Building project '/src/solution/src/subProject/tsconfig.json'...
+
+[[90m12:00:00 AM[0m] Project 'src/solution/src/subProject2/tsconfig.json' is out of date because output file 'src/solution/lib/src/subProject2/index.js' does not exist
+
+[[90m12:00:00 AM[0m] Building project '/src/solution/src/subProject2/tsconfig.json'...
+
+exitCode:: ExitStatus.Success
+
+
+//// [/src/solution/lib/src/common/nominal.d.ts]
+/// <reference types="types" />

```

Comment:
Shouldn't it be `reference path` instead? How `<reference types="types" />` will be resolved by another project referencing to this one? I mean, afaik `reference types` means "load types from types-like package (from both `typesRoot` or `node_modules/@types`) with specific name", but this is the local file, why we should refer to it with `types instead of `path`?